.github/workflows/pr_test.yml

```diff
@@ -2,25 +2,17 @@ name: "Go unit testing"
 
 on:
   pull_request:
-    types: [ assigned, opened, synchronize, reopened ]
     branches:
       - main
 
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
+      - uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0
       
-      - uses: actions/setup-go@v2
+      - uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
```

Comment:
Why are we using hashes vs versions? I there something wrong with v2?

Reply:
This is a new security ask as they have validated these specific SHAs.